db_setup.py

```diff
@@ -1,36 +1,47 @@
-import os
-import sys
-from sqlalchemy import Column, ForeignKey, Integer, String, DateTime, func, Float
-from sqlalchemy.ext.declarative import declarative_base
-from sqlalchemy.orm import relationship
-from sqlalchemy import create_engine
-
-Base = declarative_base()
-
-
-class Violation(Base):
-    __tablename__ = 'violation'
-
-    id = Column(Integer, primary_key=True)
-    car_id = Column(Integer)
-    longitude = Column(Float)
-    latitude = Column(Float)
-    speed = Column(Float)
-    time = Column(DateTime)
+from pymongo import MongoClient
+
+def _create_database(URI,db_name):
+    """
+        Warning: This function is intended only to be used inside the scope of this file
+        you should not use it outside
+        Usage: _create_database(URI,db_name) return a database object
+        Example: db = _create_database(URI,db_name)
+    """
+    clinet = MongoClient(URI)
+    return clinet[db_name]
+
+class data_manager():
+    """
+        This class was a made as wrapper for PyMongo libraries to add some constrain on collection names and to hide mango layer to be replaceable 
+        Usage:
+        Example:
+    """
+    def __init__(self,URI="mongodb://localhost:27017/",
+                 db_name="evil_car"):
+        self.db = _create_database(URI,db_name)
+
+    def _get_collection(self,collection_name):
+        return self.db[collection_name]
+
+    def add_data(self,collection_name,data):
+        """
+        """
+        # TODO: add data checkers here
+        return self._get_collection(collection_name).insert(data) 
     
-    @property
-    def serialize(self):
-
-        return {
-            'car_id' : self.car_id,
-            'longitude' : self.longitude,
-            'latitude' : self.latitude,
-            'speed' : self.speed,
-            'time' : self.time
-        }
-
-
-engine = create_engine('sqlite:///violations.db')
-
-
-Base.metadata.create_all(engine)
+    def get_data(self,collection_name,collection_filter=None):
+        """
+        """
+        col = self._get_collection(collection_name)
+        if collection_filter is not None:
+            return col.find(collection_filter,{"_id":0}) 
+        return col.find({},{"_id":0})
```

Comment:
why exclude the id 
won't we need it if wanna to edit or something ?

idk shouldn't u remove it when sending data not from here

Reply:
it was made for visualization purposes, i think will need it only if we are going to provide any deleting mechanism because (in my opinion) violations and obstacles data shouldn't be edited
but this doesn't mean to completely ignore the id and i will try to make add that feature